src/include/miopen/find_solution.hpp

```diff
@@ -182,6 +182,55 @@ struct SolverContainer
         return ss;
     }
 
+    // Search for all applicable solutions among many solvers
+    template <class Problem, class Solution = miopen::solver::ConvSolution>
+    std::vector<Solution>
+    SearchForSolutions(const ExecutionContext& ctx,
+                       const Problem& problem,
+                       std::size_t limit = std::numeric_limits<std::size_t>::max()) const
+    {
+        std::vector<Solution> ss;
+        std::size_t count    = 0;
+        const auto find_only = GetEnvFindOnlySolver();
+        miopen::each_args(
+            [&](auto solver) {
+                if(count >= limit)
+                    return;
+                if(find_only.IsValid() && find_only != Id{SolverDbId(solver)})
+                { // Do nothing (and keep silence for the sake of Tuna), just skip.
+                }
+                // For better performance, check IsDynamic() first, because
+                // it is much faster than IsApplicable().
+                // else if(problem.use_dynamic_solutions_only && !solver.IsDynamic())
+                //    MIOPEN_LOG_I2(SolverDbId(solver) << ": Skipped (non-dynamic)");
```

Comment:
[future] I think we need to keep the notion of "dynamic" solutions for all primitives.

Reply:
@DrizztDoUrden ok?

Reply:
Yeah, probably